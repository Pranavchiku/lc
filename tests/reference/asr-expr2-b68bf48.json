{
    "basename": "asr-expr2-b68bf48",
    "cmd": "lc --asr-dump --no-color {infile} -o {outfile} -extra-arg=\"-Isrc/runtime/include\"",
    "infile": "tests/expr2.c",
    "infile_hash": "11cb114715d67a4763454167abc900089b2119ecf18a980d18a30322",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr2-b68bf48.stdout",
    "stdout_hash": "2455a47a75aa01ef551061d2f90a63011dc2351e73fd89d7a9b8baaf",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}