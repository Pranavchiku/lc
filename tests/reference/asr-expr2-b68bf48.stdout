(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        3
                        {
                            __return_var:
                                (Variable
                                    3
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x:
                                (Variable
                                    3
                                    x
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(=
                        (Var 3 x)
                        (IntegerBinOp
                            (IntegerBinOp
                                (IntegerConstant 2 (Integer 4))
                                Add
                                (IntegerConstant 3 (Integer 4))
                                (Integer 4)
                                ()
                            )
                            Mul
                            (IntegerConstant 5 (Integer 4))
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 3 x)]
                        ()
                        ()
                    )
                    (=
                        (Var 3 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 3 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        4
                        {
                            exit_code:
                                (Variable
                                    4
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(=
                        (Var 4 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
